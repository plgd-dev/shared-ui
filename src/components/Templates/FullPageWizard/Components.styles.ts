import { css } from '@emotion/react'
import { ThemeType, getThemeColor, getTheme } from '../../Atomic/_theme'
import { fontPrimary, fontSecondary } from '../../Atomic/_utils/commonStyles'

export const description = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-size: 16px;
    font-style: normal;
    font-weight: 400;
    line-height: 24px;
    color: ${getThemeColor(theme, `FullPageWizard.description.color`)};
    margin: 0 0 32px 0;
`

export const descriptionLarge = css`
    margin: 0 0 48px 0;
`

export const subHeadline = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontSecondary`, fontSecondary)};
    color: ${getThemeColor(theme, `FullPageWizard.subHeadline.color`)};
    font-size: 24px;
    font-style: normal;
    font-weight: 700;
    line-height: 30px;
    letter-spacing: -0.5px;
    border-top: 1px solid ${getThemeColor(theme, `FullPageWizard.subHeadline.borderColor`)};
    padding-top: 32px;
    margin: 0 0 4px 0;
`

export const noBorder = css`
    border-top: none;
`

export const separator = (theme: ThemeType) => css`
    border-top: 1px solid ${getThemeColor(theme, `FullPageWizard.subHeadline.borderColor`)};
`

export const headline = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontSecondary`, fontSecondary)};
    color: ${getThemeColor(theme, `FullPageWizard.headline.color`)};
    font-size: 48px;
    font-style: normal;
    font-weight: 700;
    line-height: 54px;
    letter-spacing: -1.5px;
    margin: 0 0 24px 0;
`

export const groupHeadline = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontSecondary`, fontSecondary)};
    color: ${getThemeColor(theme, `FullPageWizard.groupHeadline.color`)};
    font-size: 16px;
    font-style: normal;
    font-weight: 700;
    line-height: 150%;
    letter-spacing: -0.5px;
`

export const groupHeadlineMargin = css`
    margin: 32px 0 16px 0;
`

export const flex = css`
    display: flex;
    align-items: center;
    justify-content: space-between;
`

export const expander = (theme: ThemeType) => css`
    display: flex;
    align-items: center;
    gap: 4px;
    color: ${getThemeColor(theme, `FullPageWizard.expander.color`)};
    cursor: pointer;
    transition: color 0.3s;

    &:hover {
        color: ${getThemeColor(theme, `FullPageWizard.expander.hover.color`)};
    }
`

export const expanderMargin = css`
    margin-top: 20px;
`
