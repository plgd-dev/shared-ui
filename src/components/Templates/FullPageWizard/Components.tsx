import React, { FC, useState } from 'react'
import * as styles from './Components.styles'
import ConditionalWrapper from '../../Atomic/ConditionalWrapper'
import Tooltip from '../../Atomic/Tooltip'
import IconQuestion from '../../Atomic/Icon/components/IconQuestion'
import { convertSize } from '../../Atomic/Icon'
import ShowAnimate from '../../Atomic/ShowAnimate'
import IconSettings from '../../Atomic/Icon/components/IconSettings'

export type DescriptionProps = {
    children: string
    large?: boolean
}

export type SubHeadlineProps = {
    children: string
    noBorder?: boolean
}

export type HeadlineProps = {
    children: string
}

export type GroupHeadlineProps = {
    children: string
    noBorder?: boolean
    tooltipText?: string
    tooltipMaxWidth?: number
}

export type ToggleConfigurationProps = {
    defaultShow?: boolean
    children: React.ReactNode | React.ReactNode[]
    i18n: {
        hide: string
        show: string
    }
    margin?: boolean
}

export const Description: FC<DescriptionProps> = (props) => <p css={[styles.description, props.large && styles.descriptionLarge]}>{props.children}</p>

export const SubHeadline: FC<SubHeadlineProps> = (props) => <h2 css={[styles.subHeadline, props.noBorder && styles.noBorder]}>{props.children}</h2>

export const Headline: FC<HeadlineProps> = (props) => <h1 css={[styles.headline]}>{props.children}</h1>

export const GroupHeadline: FC<GroupHeadlineProps> = (props) => (
    <ConditionalWrapper
        condition={!!props.tooltipText}
        wrapper={(children) => (
            <div css={[styles.flex, styles.groupHeadlineMargin]}>
                <span>{children}</span>
                <Tooltip content={props.tooltipText} delay={200} maxWidth={props.tooltipMaxWidth || 300} portalTarget={document.getElementById('modal-root')!}>
                    <IconQuestion {...convertSize(16)} />
                </Tooltip>
            </div>
        )}
    >
        <h3 css={[styles.groupHeadline, !props.tooltipText && styles.groupHeadlineMargin]}>{props.children}</h3>
    </ConditionalWrapper>
)

export const ToggleConfiguration: FC<ToggleConfigurationProps> = (props) => {
    const { i18n, children, margin = true } = props
    const [show, setShow] = useState(false)

    return (
        <>
            <ShowAnimate show={show}>
                <>{children}</>
            </ShowAnimate>
            <div css={[styles.expander, margin && styles.expanderMargin]} onClick={() => setShow(!show)}>
                <IconSettings /> {show ? i18n.hide : i18n.show}
            </div>
        </>
    )
}
