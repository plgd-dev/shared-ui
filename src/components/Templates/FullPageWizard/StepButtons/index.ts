export { default } from './StepButtons'
export * from './StepButtons'
