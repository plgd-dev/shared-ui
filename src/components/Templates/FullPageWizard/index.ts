export { default } from './FullPageWizard'
export * from './FullPageWizard'
