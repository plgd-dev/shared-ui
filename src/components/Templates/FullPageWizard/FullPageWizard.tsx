import React from 'react'
import { Helmet } from 'react-helmet'
import { useTheme } from '@emotion/react'
import isFunction from 'lodash/isFunction'

import * as styles from './FullPageWizard.styles'
import { ThemeType } from '../../Atomic/_theme'
import Logo from '../../Atomic/Logo'
import IconCircle from './assets/IconCircle'
import IconDone from './assets/IconDone'
import IconCloseCircle from '../../Atomic/Icon/components/IconCloseCircle'
import { convertSize } from '../../Atomic'
import { ComponentType, Props } from './FullPageWizard.types'
import { Description, SubHeadline } from './Components'

const FullPageWizard: ComponentType<Props> = (props) => {
    const { children, title, i18n, steps, activeStep, onStepChange, onClose } = props

    const theme: ThemeType = useTheme()

    return (
        <div css={styles.wizard}>
            <Helmet>
                <title>{title}</title>
            </Helmet>
            <div css={styles.row}>
                <div css={styles.leftCol}>
                    <div css={styles.leftInner}>
                        {theme.logo && <Logo logo={theme.logo} />}

                        <ul css={styles.navigation}>
                            {steps.map((step, key) => (
                                <li css={[activeStep >= key && styles.visited]} key={key}>
                                    <div css={styles.icon}>
                                        {activeStep > key ? (
                                            <IconDone fill={theme.colorPalette.primary} />
                                        ) : (
                                            <IconCircle fill={activeStep >= key ? theme.colorPalette.primary : theme.colorPalette.neutral300} />
                                        )}
                                    </div>
                                    <a
                                        css={[styles.link, activeStep >= key && styles.visited]}
                                        href='#'
                                        onClick={(e) => {
                                            e.preventDefault()
                                            isFunction(onStepChange) && onStepChange(key)
                                        }}
                                    >
                                        {step.name}
                                    </a>
                                    {step.description && <p css={[styles.description, styles.visited]}>{step.description}</p>}
                                </li>
                            ))}
                        </ul>
                    </div>
                </div>
                <div css={styles.rightInner}>
                    <a
                        css={styles.close}
                        href='#'
                        onClick={(e) => {
                            e.preventDefault()
                            isFunction(onClose) && onClose()
                        }}
                    >
                        <span>{i18n.close}</span>
                        <IconCloseCircle {...convertSize(32)} />
                    </a>
                    <div css={styles.content}>{children}</div>
                </div>
            </div>
        </div>
    )
}

FullPageWizard.displayName = 'FullPageWizard'
FullPageWizard.Description = Description
FullPageWizard.SubHeadline = SubHeadline

export default FullPageWizard
