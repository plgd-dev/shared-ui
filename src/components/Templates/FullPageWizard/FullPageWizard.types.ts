import { FC, ReactNode } from 'react'
import { DescriptionProps, HeadlineProps, SubHeadlineProps, GroupHeadlineProps, ToggleConfigurationProps } from './Components'

export interface ComponentType<P> extends FC<P> {
    Description: FC<DescriptionProps>
    Headline: FC<HeadlineProps>
    SubHeadline: FC<SubHeadlineProps>
    GroupHeadline: FC<GroupHeadlineProps>
    ToggleConfiguration: FC<ToggleConfigurationProps>
}

export type Props = {
    activeStep: number
    children?: ReactNode
    i18n: {
        close: string
    }
    onClose: () => void
    onStepChange: (step: number) => void
    steps: { name: string; description?: string; link?: string }[]
    title: string
    visitedStep?: number
}
