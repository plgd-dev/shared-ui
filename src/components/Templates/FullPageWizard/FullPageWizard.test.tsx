import React from 'react'
import { fireEvent } from '@testing-library/react'

import { render } from '../../../../test/jest-wrapper'
import FullPageWizard from './FullPageWizard'

describe('FullPageWizard', () => {
    const props = {
        activeStep: 0,
        steps: [
            { name: 'Step 1', description: 'Description 1' },
            { name: 'Step 2', description: 'Description 2' },
            { name: 'Step 3', description: 'Description 3' },
        ],
        onStepChange: () => {},
        title: 'Test Title',
    }

    it('renders without crashing', () => {
        const { container, asFragment } = render(<FullPageWizard {...props} />)
        expect(container.firstChild).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders the correct number of steps', () => {
        const { getAllByRole } = render(<FullPageWizard {...props} />)
        const steps = getAllByRole('listitem')
        expect(steps.length).toBe(props.steps.length)
    })

    it('calls onStepChange when a step is clicked', () => {
        const mockOnStepChange = jest.fn()
        const { getAllByRole } = render(<FullPageWizard {...props} activeStep={1} onStepChange={mockOnStepChange} visitedStep={0} />)
        const steps = getAllByRole('link')
        fireEvent.click(steps[0])
        expect(mockOnStepChange).toHaveBeenCalledWith(0)
    })

    it('does not call onStepChange when a non-visited step is clicked', () => {
        const mockOnStepChange = jest.fn()
        const { getAllByRole } = render(<FullPageWizard {...props} onStepChange={mockOnStepChange} visitedStep={0} />)
        const steps = getAllByRole('link')
        fireEvent.click(steps[1])
        expect(mockOnStepChange).not.toHaveBeenCalled()
    })

    it('calls onClose when close button is clicked', () => {
        const mockOnClose = jest.fn()
        const { getByRole } = render(<FullPageWizard {...props} onClose={mockOnClose} />)
        const closeButton = getByRole('button')
        fireEvent.click(closeButton)
        expect(mockOnClose).toHaveBeenCalled()
    })
})
