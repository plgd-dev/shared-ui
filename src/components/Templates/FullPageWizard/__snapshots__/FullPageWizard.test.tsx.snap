// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPageWizard renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  background: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow: hidden;
}

.emotion-2 {
  background: #F4F9FB;
  -webkit-flex: 0 0 500px;
  -ms-flex: 0 0 500px;
  flex: 0 0 500px;
}

@media (max-width: 1199px) {
  .emotion-2 {
    -webkit-flex: 0 0 400px;
    -ms-flex: 0 0 400px;
    flex: 0 0 400px;
  }
}

@media (max-width: 991px) {
  .emotion-2 {
    -webkit-flex: 0 0 320px;
    -ms-flex: 0 0 320px;
    flex: 0 0 320px;
  }
}

.emotion-3 {
  padding: 32px 48px;
}

@media (max-width: 991px) {
  .emotion-3 {
    padding: 16px;
  }
}

.emotion-5 {
  margin: 64px 0 0 0;
  padding-left: 56px;
  list-style: none;
  position: relative;
}

.emotion-5 li {
  display: block;
  position: relative;
}

.emotion-5 li+li {
  margin-top: 24px;
}

.emotion-5 li+li:after {
  content: '';
  display: block;
  background: #D7D8DA;
  width: 1px;
  height: calc(100% - 20px);
  position: absolute;
  left: -28px;
  top: -20px;
  bottom: 10px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.emotion-6 {
  opacity: 1;
}

.emotion-6:after {
  background: #2261AE!important;
}

.emotion-7 {
  width: 32px;
  height: 32px;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  left: -44px;
  background: #F4F9FB;
}

.emotion-8 {
  color: #0A2965;
  font-weight: 600;
  opacity: 0.6;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: default;
  opacity: 1;
}

.emotion-8:hover {
  color: #0A2965;
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
}

.emotion-8:after {
  background: #2261AE!important;
}

.emotion-9 {
  color: #81868C;
  font-size: 12px;
  font-weight: 400;
  line-height: 18px;
  margin: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  opacity: 1;
}

.emotion-9:after {
  background: #2261AE!important;
}

.emotion-12 {
  color: #0A2965;
  font-weight: 600;
  opacity: 0.6;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: default;
}

.emotion-12:hover {
  color: #0A2965;
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
}

.emotion-18 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 32px 48px;
  position: relative;
  overflow: auto;
}

@media (max-width: 991px) {
  .emotion-18 {
    padding: 16px;
  }
}

.emotion-19 {
  position: fixed;
  top: 28px;
  right: 44px;
  color: #81868C;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  padding: 4px;
  z-index: 2;
  border-radius: 8px;
}

.emotion-19 span {
  padding-right: 8px;
}

@media (max-width: 991px) {
  .emotion-19 span {
    display: none;
  }
}

.emotion-19:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #2261AE;
}

.emotion-20 {
  max-width: 600px;
  margin: auto;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <img
            alt=""
            class="emotion-4"
            height="32px"
            id="main-app-logo"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQ3IiBoZWlnaHQ9IjMyIiB2aWV3Qm94PSIwIDAgMTQ3IDMyIiBmaWxsPSJub25lIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHJvbGU9ImltZyIgY2xhc3M9ImNzcy1qbHBlb3UiPjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNODUuMTA2IDExLjgzNWMxLjQ5NCAwIDIuNjMuNDg1IDMuNDA4IDEuNDU3di0xLjMwNWgyLjgzNnY4LjQ4M2MwIDEuODY3LS40OTMgMy4yNTgtMS40NzYgNC4xNjgtLjk4NC45MDctMi40MiAxLjM2Mi00LjMwOCAxLjM2Mi0uOTk2IDAtMS45NDItLjEyLTIuODM0LS4zNi0uODk1LS4yNC0xLjYzNi0uNTg4LTIuMjIyLTEuMDQyLjExOS0uMjEgMS4wNy0xLjkwNyAxLjE4OC0yLjEyLjQzMi4zNTUuOTguNjM1IDEuNjQ2Ljg0Mi42NjUuMjEgMS4zMjcuMzEyIDEuOTkyLjMxMiAxLjAzNCAwIDEuNzk3LS4yMjkgMi4yODgtLjY5LjQ5My0uNDYyLjczOC0xLjE2LjczOC0yLjA5NXYtLjQzNGMtLjc3OS44NDctMS44NjYgMS4yNy0zLjI1NiAxLjI3YTUuNTMgNS41MyAwIDAgMS0yLjU5Ni0uNjE1IDQuNzUzIDQuNzUzIDAgMCAxLTEuODY2LTEuNzM2Yy0uNDYtLjc0Mi0uNjktMS42MDItLjY5LTIuNTc0IDAtLjk3My4yMy0xLjgzLjY5LTIuNTc1YTQuNzI1IDQuNzI1IDAgMCAxIDEuODY2LTEuNzMzYy43ODYtLjQxIDEuNjUtLjYxNSAyLjU5Ni0uNjE1Wm0tMTkuMDE2LS4wMDNjLjk1OCAwIDEuODMuMjE4IDIuNjE3LjY1NWE0LjcxIDQuNzEgMCAwIDEgMS44NDUgMS44MzVjLjQ0Ny43OS42NzIgMS43MDguNjcyIDIuNzU4IDAgMS4wNDctLjIyNSAxLjk2NS0uNjcyIDIuNzU1YTQuNzE5IDQuNzE5IDAgMCAxLTEuODQ1IDEuODM4Yy0uNzg3LjQzNS0xLjY1OS42NTItMi42MTcuNjUyLTEuMzE0IDAtMi4zNDktLjQxLTMuMTAyLTEuMjMzdjQuNzU2SDYwVjExLjk4M2gyLjg1NHYxLjE3NWMuNzQtLjg4MyAxLjgyLTEuMzI1IDMuMjM2LTEuMzI1Wm02My44NzEgMGMuOTgzIDAgMS44NzYuMjA1IDIuNjguNjE1YTQuNzgzIDQuNzgzIDAgMCAxIDEuOTI2IDEuNzgzYy40NzguNzc1LjcxOCAxLjY5NS43MTggMi43NTVsLTcuNjIyIDEuNDU4Yy4yMTcuNTA1LjU1OS44ODQgMS4wMjQgMS4xMzcuNDY1LjI1MyAxLjAzNy4zNzggMS43MTQuMzc4LjUzNiAwIDEuMDExLS4wNzggMS40MjgtLjIzNWEzLjQ4IDMuNDggMCAwIDAgMS4xNTgtLjc1bDEuNTkgMS43MDVjLS45NzEgMS4wOTctMi4zODkgMS42NDctNC4yNTIgMS42NDctMS4xNjMgMC0yLjE5MS0uMjIyLTMuMDg0LS42Ny0uODk1LS40NS0xLjU4My0xLjA3My0yLjA2OC0xLjg2Ny0uNDg1LS43OTYtLjcyOC0xLjY5OC0uNzI4LTIuNzA4IDAtLjk5OC4yNC0xLjg5Ny43MTgtMi43YTUuMDA0IDUuMDA0IDAgMCAxIDEuOTcyLTEuODc1Yy44MzctLjQ0OCAxLjc3OS0uNjczIDIuODI2LS42NzNabS03LjQyMi0zLjcxMnYxNC4wNTVoLTIuODU0VjIxYy0uNzQxLjg4NS0xLjgxNiAxLjMyNS0zLjIxOSAxLjMyNS0uOTcxIDAtMS44NDgtLjIxNS0yLjYzNC0uNjQyYTQuNjU3IDQuNjU3IDAgMCAxLTEuODQ4LTEuODM4Yy0uNDQ3LS43OTUtLjY3LTEuNzE4LS42Ny0yLjc2NSAwLTEuMDQ3LjIyNS0xLjk3LjY3LTIuNzY1YTQuNjU4IDQuNjU4IDAgMCAxIDEuODQ4LTEuODM3Yy43ODYtLjQzIDEuNjYzLS42NDYgMi42MzQtLjY0NiAxLjMxNSAwIDIuMzQ0LjQxMyAzLjA4NSAxLjIzM1Y4LjEyaDIuOTg4Wm0tMTQuMjk3IDEwLjUzYy41MjQgMCAuOTY0LjE3IDEuMzIzLjUwMy4zNTYuMzM0LjUzNi43NzUuNTM2IDEuMzE3IDAgLjUzLS4xOC45NzMtLjUzNiAxLjMyNi0uMzU5LjM1NC0uNzk5LjUzLTEuMzIzLjUzLS41MjMgMC0uOTY1LS4xNzYtMS4zMjItLjUzLS4zNTYtLjM1NC0uNTM2LS43OTYtLjUzNi0xLjMyNiAwLS41NDIuMTgtLjk4My41MzYtMS4zMTcuMzU3LS4zMzMuNzk5LS41MDMgMS4zMjItLjUwM1pNNzYuMjAyIDguMTJ2MTAuNTEyYzAgLjQzLjExNC43Ni4zMzcuOTk2LjIyMi4yMzIuNTM4LjM1Ljk0Ny4zNS4xNTQgMCAuMzA5LS4wMi40Ny0uMDU4LjE2LS4wMzcuMjg0LS4wODMuMzcyLS4xMzMuMDE1LjIzLjEyMiAyLjA2My4xMzQgMi4yOTNhNC45NiA0Ljk2IDAgMCAxLTEuNTUuMjQ1Yy0xLjE2MiAwLTIuMDctLjMwNS0yLjcyLS45MTctLjY1Mi0uNjEzLS45NzUtMS40OC0uOTc1LTIuNjA1VjguMTJoMi45ODVaTTEwNC40NzEgOHYxNC4wNTVoLTIuODU0VjIwLjg4Yy0uNzQxLjg4NS0xLjgxMyAxLjMyNS0zLjIxOSAxLjMyNS0uOTY4IDAtMS44NDgtLjIxMi0yLjYzMi0uNjQyYTQuNjUzIDQuNjUzIDAgMCAxLTEuODQ4LTEuODM4Yy0uNDQ3LS43OTUtLjY3LTEuNzE4LS42Ny0yLjc2NSAwLTEuMDQ3LjIyMy0xLjk3LjY3LTIuNzY1YTQuNjQ2IDQuNjQ2IDAgMCAxIDEuODQ4LTEuODM4Yy43ODQtLjQzIDEuNjY0LS42NDIgMi42MzItLjY0MiAxLjMxNyAwIDIuMzQ0LjQxIDMuMDg1IDEuMjNWOGgyLjk4OFptMzQuMDg4IDMuOTQzIDIuNzMgNy42NzcgMi43My03LjY3N0gxNDdsLTQuMDk4IDEwLjIyNWgtMy4yODRsLTQuMDk4LTEwLjIyNWgzLjAzOVptLTIxLjU3NyAyLjMxNWMtLjc2NiAwLTEuMzk4LjI1NS0xLjg5Ni43NjctLjQ5OC41MS0uNzQ2IDEuMTk1LS43NDYgMi4wNTUgMCAuODU4LjI0OCAxLjU0Mi43NDYgMi4wNTUuNS41MSAxLjEzLjc2NSAxLjg5Ni43NjUuNzU2IDAgMS4zOC0uMjU1IDEuODc4LS43NjUuNDk5LS41MTMuNzQ2LTEuMTk3Ljc0Ni0yLjA1NSAwLS44Ni0uMjQ3LTEuNTQ1LS43NDYtMi4wNTUtLjQ5OC0uNTEyLTEuMTI1LS43NjctMS44NzgtLjc2N1ptLTUxLjQwOCAwYy0uNzY2IDAtMS4zOTUuMjU1LTEuODg2Ljc2Ny0uNDkzLjUxLS43MzggMS4xOTUtLjczOCAyLjA1NSAwIC44NTguMjQ1IDEuNTQyLjczOCAyLjA1NS40OS41MSAxLjEyLjc2NSAxLjg4Ni43NjUuNzY3IDAgMS4zOTYtLjI1NSAxLjg4Ni0uNzY1LjQ5LS41MTMuNzM5LTEuMTk3LjczOS0yLjA1NSAwLS44Ni0uMjQ4LTEuNTQ1LS43NC0yLjA1NS0uNDktLjUxMi0xLjExOC0uNzY3LTEuODg1LS43NjdabTMzLjM0My0uMTJjLS43NjYgMC0xLjM5OC4yNTUtMS44OTYuNzY3LS40OTguNTEtLjc0OSAxLjE5NS0uNzQ5IDIuMDU1IDAgLjg1OC4yNSAxLjU0My43NDkgMi4wNTUuNDk4LjUxIDEuMTMuNzY1IDEuODk1Ljc2NS43NTQuMDAzIDEuMzgxLS4yNTUgMS44NzYtLjc2NS40OTktLjUxMi43NDktMS4xOTcuNzQ5LTIuMDU1IDAtLjg2LS4yNS0xLjU0NS0uNzQ5LTIuMDU1LS40OTgtLjUxMi0xLjEyMi0uNzY4LTEuODc2LS43NjhabS0xMy4yMTcuMTJjLS43OTEgMC0xLjQ0My4yMy0xLjk1NC42OTItLjUwOC40Ni0uNzYzIDEuMDYzLS43NjMgMS44MDggMCAuNzQ0LjI1NSAxLjM0Ny43NjMgMS44MS41MS40NiAxLjE2My42OSAxLjk1NC42OS43OTEgMCAxLjQzOC0uMjMgMS45NDQtLjY5LjUwNi0uNDYzLjc1Ni0xLjA2Ni43NTYtMS44MSAwLS43NDUtLjI1LTEuMzQ4LS43NTYtMS44MDgtLjUwNi0uNDYzLTEuMTUzLS42OTItMS45NDQtLjY5MlptNDQuMjYxLS4xOWMtLjc1MyAwLTEuMzY4LjI0LTEuODM4LjcyLS40NzMuNDgtLjcyMyAxLjE0Mi0uNzQ4IDEuOTkuNTAxLS4wOTggNC41MTUtLjg3IDUuMDE4LS45NjhhMi4yMjYgMi4yMjYgMCAwIDAtLjg2Mi0xLjI2OGMtLjQzNS0uMzE3LS45NTgtLjQ3NC0xLjU3LS40NzRaIiBmaWxsPSIjMTkxQTFBIj48L3BhdGg+PHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00Ny40MDUgMjcuMDA5QTEyLjY4IDEyLjY4IDAgMCAwIDUwIDE5LjMwMmMwLTcuMDA3LTUuNjQ2LTEyLjY4OC0xMi42MS0xMi42ODgtLjU2NiAwLTEuMTIyLjA0Mi0xLjY2OS4xMTUtLjQ1LS43MjMtLjgwNi0xLjI2NS0xLjAyNC0xLjU0NEMzMi4yMzMgMi4wMjcgMjguNDA1IDAgMjQuMTA2IDBjLTYuMzcgMC0xMS43MDQgNC40NTYtMTMuMDk4IDEwLjQzOCAwLS4wMDgtLjAwOC0uMDE4LS4wMDYtLjAyNS0uMS0uMDAzLS4xOTktLjAxLS4zLS4wMUM0Ljc5MyAxMC40MDMgMCAxNS4yMjYgMCAyMS4xNzMgMCAyNy4xMTggNC43OTIgMzEuOTQgMTAuNzAzIDMxLjk0YzMuOTM2IDAgOS4yOC4wNDcgMTIuOTguMDQ3IDIuMjA3IDAgMy40MDItLjAyNiAxMi41MTMgMGwuMjItLjAwMmMxLjE4NC4wNTYgMy43MjkuMDExIDYuMTQ3LTEuMTA5LjM4OC0uMTg0IDIuNzQ1LTEuMTc2IDQuODQzLTMuODY2bC0uMDAxLS4wMDFaIiBmaWxsPSIjRkVCRjQwIj48L3BhdGg+PHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00Ny40MDUgMjcuMDA5QTEyLjY4IDEyLjY4IDAgMCAwIDUwIDE5LjMwMmMwLTcuMDA3LTUuNjQ2LTEyLjY4OC0xMi42MS0xMi42ODgtLjU2NiAwLTEuMTIyLjA0Mi0xLjY2OC4xMTUtLjQ1LS43MjMtLjgwNi0xLjI2NS0xLjAyNS0xLjU0NEMzMi4yMzMgMi4wMjcgMjguNDA2IDAgMjQuMTA2IDBjLTcuNDMgMC0xMy40NTcgNi4wNjEtMTMuNDU3IDEzLjU0IDAgNy40NzggNS42MDIgMTguNDQ3IDEzLjAzNCAxOC40NDcgMi4yMDcgMCAzLjQwMy0uMDI2IDEyLjUxMyAwbC4yMjEtLjAwMmMxLjE4My4wNTYgMy43MjguMDExIDYuMTQ3LTEuMTA5LjM4OC0uMTg0IDIuNzQ0LTEuMTc2IDQuODQyLTMuODY2bC0uMDAxLS4wMDFaIiBmaWxsPSIjODdDRUYyIj48L3BhdGg+PHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00Ni45MyAyNi4yOTZDNDQuODgyIDIyLjYyMiAzNi4yNzggNy4yMSAzNC42OTYgNS4xODQgMzIuMjMzIDIuMDI3IDI4LjQwNCAwIDI0LjEwNSAwYy03LjQzIDAtMTMuNDU3IDYuMDYxLTEzLjQ1NyAxMy41NCAwIDcuNDc4IDUuNjAyIDE4LjQ0NyAxMy4wMzQgMTguNDQ3IDIuMjA4IDAgMy40MDMtLjAyNiAxMi41MTMgMGwuMjIxLS4wMDJjMS4xODQuMDU2IDMuNzI5LjAxMiA2LjE0Ny0xLjEwOS4zODgtLjE4NCAyLjc0NS0xLjE3NiA0Ljg0Mi0zLjg2NmwtLjQ3Ni0uNzE0WiIgZmlsbD0iIzIyNjFBRSI+PC9wYXRoPjwvc3ZnPgo="
            width="147px"
          />
          <ul
            class="emotion-5"
          >
            <li
              class="emotion-6"
            >
              <div
                class="emotion-7"
              >
                <svg
                  fill="none"
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="16"
                    r="6.25"
                    stroke="#2261AE"
                    stroke-width="1.5"
                  />
                </svg>
              </div>
              <a
                class="emotion-8"
                href="#"
              >
                Step 1
              </a>
              <p
                class="emotion-9"
              >
                Description 1
              </p>
            </li>
            <li
              class="emotion-10"
            >
              <div
                class="emotion-7"
              >
                <svg
                  fill="none"
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="16"
                    r="6.25"
                    stroke="#D7D8DA"
                    stroke-width="1.5"
                  />
                </svg>
              </div>
              <a
                class="emotion-12"
                href="#"
              >
                Step 2
              </a>
              <p
                class="emotion-9"
              >
                Description 2
              </p>
            </li>
            <li
              class="emotion-10"
            >
              <div
                class="emotion-7"
              >
                <svg
                  fill="none"
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="16"
                    r="6.25"
                    stroke="#D7D8DA"
                    stroke-width="1.5"
                  />
                </svg>
              </div>
              <a
                class="emotion-12"
                href="#"
              >
                Step 3
              </a>
              <p
                class="emotion-9"
              >
                Description 3
              </p>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="emotion-18"
      >
        <a
          class="emotion-19"
          href="#"
          role="button"
        >
          <span />
          <svg
            fill="none"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 29.333c7.333 0 13.333-6 13.333-13.333S23.333 2.667 16 2.667 2.667 8.667 2.667 16s6 13.333 13.333 13.333M12.227 19.773l7.546-7.546M19.773 19.773l-7.546-7.546"
              stroke="currentcolor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </a>
        <div
          class="emotion-20"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
